java/google/registry/whois/WhoisCommandFactory.java

```diff
@@ -0,0 +1,53 @@
+// Copyright 2016 The Nomulus Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package google.registry.whois;
+
+import com.google.common.net.InternetDomainName;
+import google.registry.config.RegistryConfig.ConfigModule;
+
+import javax.annotation.Nullable;
+import java.net.InetAddress;
+
+/**
+ * A no-op base class used to provide custom whois commands.
+ * <p>To add custom commands, extend this class, then configure it in
+ * {@link ConfigModule#provideWhoisCommandFactoryClass()}.
+ */
+public class WhoisCommandFactory {
+
+  public WhoisCommand domainLookupCommand(InternetDomainName domainName) {
```

Comment:
Hey Ben,

Maybe I'm misunderstanding where you would like to see the custom logic injection, but the reason we overrode the commands is due to their simplicity. Each command does the same thing, lookup an object in Datastore then checks if the object was found. Adding injection points (before, after, etc) when the commands are so simple doesn't seem like it provides any real benefit. In our use case we only needed to add a single validation line after DomainLookupCommand#getSuccessResponse. This makes our command a whole two lines, call the parent method then validate.

I also went the the factory route over overriding as we will likely need additional custom commands in the future. Instead of putting a temporary hack in place by coping a bunch of code, I've made the change in anticipation of that future work.